src/main.ts

```diff
@@ -1125,11 +1125,29 @@ function generateEncode(ctx: Context, fullName: string, messageDesc: DescriptorP
         }
       `);
     } else if (isScalar(field) || isEnum(field)) {
-      chunks.push(code`
-        if (${notDefaultCheck(ctx, field, messageDesc.options, `message.${fieldName}`)}) {
```

Comment:
Hey @khi195 ; thanks for the PR! Instead of adding more options, I think we should try and solve this by being smarter at handling "there is no enum with value === 0".

I.e. this line is calling `notDefaultCheck`, which inside of that method has some code that looks like:

```
      const zerothValue = enumProto.value.find((v) => v.number === 0) || enumProto.value[0];
      if (options.stringEnums) {
        const enumType = messageToTypeName(ctx, field.typeName);
        return code`${maybeNotUndefinedAnd} ${place} !== ${enumType}.${zerothValue.name}`;
      } else {
        return code`${maybeNotUndefinedAnd} ${place} !== ${zerothValue.number}`;
      }
```

Note the `|| enumProto.value[0]`. That is the line that is tripping up your code, b/c it means ts-proto is treating the "the enum value 1" as the default, which it just shouldn't do.

Can you try removing that `|| enumProto.value[0]` and instead doing something like:

```
      const zerothValue = enumProto.value.find((v) => v.number === 0);
      if (!zerothValue) {
        return undefined;
      } else if (options.stringEnums) {
        const enumType = messageToTypeName(ctx, field.typeName);
        return code`${maybeNotUndefinedAnd} ${place} !== ${enumType}.${zerothValue.name}`;
      } else {
        return code`${maybeNotUndefinedAnd} ${place} !== ${zerothValue.number}`;
      }
```

And then here in the `generateEncode` method, you could do:

```
const maybeDefaultCheck = notDefaultCheck(ctx, field, messageDesc.options, `message.${fieldName}`);
```

And if `maybeDefaultCheck` is undefined, then leave the `if` statement out, just like you're currently doing by looking at the new `disableDefaultCheck` options.

I'm pretty sure that should work.